src/hotspot/share/opto/c2_globals.hpp

```diff
@@ -354,6 +354,12 @@
   notproduct(bool, TraceNewVectors, false,                                  \
           "Trace creation of Vector nodes")                                 \
                                                                             \
+  product(bool, MergeStores, true, DIAGNOSTIC,                              \
+          "Optimize stores by combining values into larger store")          \
+                                                                            \
+  notproduct(bool, TraceMergeStores, false,                                 \
+          "Trace creation of merged stores")                                \
+                                                                            \
```

Comment:
The flag should be `develop` since it is under `#ifdef ASSERT`.

Reply:
I think I'll just move the flag under NOT_PRODUCT